recipes/boost/all/test_package/conanfile.py

```diff
@@ -9,26 +9,27 @@ class DefaultNameConan(ConanFile):
     generators = "cmake"
 
     def build(self):
-        cmake = CMake(self)
-        if self.options["boost"].header_only:
-            cmake.definitions["HEADER_ONLY"] = "TRUE"
-        else:
-            cmake.definitions["Boost_USE_STATIC_LIBS"] = not self.options["boost"].shared
-        if not self.options["boost"].without_python:
-            cmake.definitions["WITH_PYTHON"] = "TRUE"
-        if not self.options["boost"].without_random:
-            cmake.definitions["WITH_RANDOM"] = "TRUE"
-        if not self.options["boost"].without_regex:
-            cmake.definitions["WITH_REGEX"] = "TRUE"
-        if not self.options["boost"].without_test:
-            cmake.definitions["WITH_TEST"] = "TRUE"
-        if not self.options["boost"].without_coroutine:
-            cmake.definitions["WITH_COROUTINE"] = "TRUE"
-        if not self.options["boost"].without_chrono:
-            cmake.definitions["WITH_CHRONO"] = "TRUE"
-        cmake.definitions["Boost_NO_BOOST_CMAKE"] = "TRUE"
-        cmake.configure()
-        cmake.build()
+        with tools.vcvars(self.settings) if (self.settings.os == "Windows" and self.settings.compiler == "clang") else tools.no_op():
```

Comment:
this shouldn't be needed - CMake helper automatically calls vcvars (if needed)

Reply:
No.. This fails when using clang-cl. Probably CMake helper didn't recognize clang-cl as a compiler requiring vcvars.

Reply:
this is very [suspicious](https://github.com/conan-io/conan/blob/e8cecad42fe83c7d8baad2d5de233d6c61304943/conans/client/build/cmake.py#L218). because we have had a logic to handle that.
could you please submit a new issue so we can take a look?
maybe providing some additional information if possible (build log, profile, generator you have used, etc.)

Reply:
See conan-io/conan#6927